src/app/pages/partners/partner-details/partner-details.js

```diff
@@ -56,23 +56,26 @@ function useRealTitles(books) {
 //     );
 // }
 
-function RequestInfoButton({infoUrl, infoText, partnerName}) {
+function RequestInfoButton({infoUrl, infoText='Request info', partnerName}) {
+    const {books} = usePartnerContext();
+    const validTitle = books.find((b) => b.length > 0); // Quirk: no books is an array of one empty string
+
     function trackInfoRequest() {
         analyticsEvents.requestInfo(partnerName);
     }
 
     // NOTE: click is not propagating up to router. For external form, target is _blank
     // For custom local form, remove that.
     return (
-        infoUrl && infoText &&
-            <section>
-                <a
-                    className="btn primary" href={infoUrl} onClick={trackInfoRequest}
-                    target="_blank" rel="noreferrer"
-                >
-                    {infoText}
-                </a>
-            </section>
+        infoUrl && validTitle &&
```

Comment:
My editor's auto-indent sometimes gets confused by JSX like this and I add `()` to wrap.  I find that clearer anyway:

```
infoUrl && validTitle && (
    <section>
        ...
    </section>
)
```